sklego/meta/decay_estimator.py

```diff
@@ -16,15 +16,27 @@ class DecayEstimator(BaseEstimator):
     This meta estimator will only work for estimators that have a
     "sample_weights" argument in their `.fit()` method.
 
+    The `fit` method computes the weights to pass to the estimator.
+
+    .. warning:: By default all the checks on `X` and `y` are delegated to the wrapped estimator.
+
+        To change such behaviour, set `check_X_y` to `True`.
+
+        Remark that if the check is skipped, then `y` should have a `shape` attrbute, which is
+        used to extract the number of samples in training data, and compute the weights.
+
     The DecayEstimator will use exponential decay to weight the parameters.
 
     w_{t-1} = decay * w_{t}
     """
 
-    def __init__(self, model, decay: float = 0.999, decay_func="exponential"):
+    def __init__(
+        self, model, decay: float = 0.999, decay_func="exponential", check_X_y=False
```

Comment:
This is a bit of a nitpick. 

```suggestion
        self, model, decay: float = 0.999, decay_func="exponential", check_input=False
```

My reasoning here is that some estimators may only have a `check_X`. This is the case when the component is a scikit-learn transformer. But it'd be nice to have a consistent name for checking the internal thing. So maybe this namechange? 

Once that change is in/discussed I think we can hit merge :)

Reply:
That's very reasonable 👌 let me adjust that!